src/fu-engine.c

```diff
@@ -304,6 +304,12 @@ fu_engine_set_release_from_appstream (FuEngine *self,
 		if (uri == NULL)
 			uri = g_strdup (tmp);
 		fwupd_release_set_uri (rel, uri);
+	} else if (remote != NULL &&
+		   fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_DIRECTORY) {
+		g_autofree gchar *uri = NULL;
+		tmp = xb_node_query_text (component, "../custom/value[@key='fwupd::FilenameCache']", NULL);
```

Comment:
Should we test for tmp==NULL and return an error in that case?

Reply:
it's a void function, but I'll add a check for `tmp != NULL`